src/content/containers/URLManager.js

```diff
@@ -78,8 +81,8 @@ URLManager.propTypes = {
 export default connect(state => ({
   urlState: state.urlState,
   isURLSetupDone: getIsURLSetupDone(state),
-}), (dispatch: Dispatch) => ({
+}), (dispatch: Dispatch<State, Action>) => ({
   updateURLState: urlState => dispatch({ type: '@@urlenhancer/updateURLState', urlState }),
   urlSetupDone: () => dispatch({ type: '@@urlenhancer/urlSetupDone' }),
-  show404: url => dispatch({ type: 'ROUTE_NOT_FOUND', url }),
+  show404: url => dispatch(({ type: 'ROUTE_NOT_FOUND', url }: Action)),
```

Comment:
Flow is unable to correctly infer what is going on here... I had to be explicit with the type.